src/url_aggregator.cpp

```diff
@@ -107,22 +183,46 @@ inline void url_aggregator::copy_scheme(const url_aggregator& u) noexcept {
   ADA_ASSERT_TRUE(validate());
 }
 
+inline void url_aggregator::set_scheme_fast(std::string_view new_scheme_with_colon) noexcept {
+  ada_log("url_aggregator::set_scheme_fast ", new_scheme_with_colon);
+  ADA_ASSERT_TRUE(validate());
+  ADA_ASSERT_TRUE(!new_scheme_with_colon.empty() && new_scheme_with_colon.back() == ':');
+  uint32_t new_difference = uint32_t(new_scheme_with_colon.size()) - components.protocol_end;
```

Comment:
If you have a `https` and later change your scheme to `ftp`, due to unsigned integer constraint, we would have a really large unsigned integer, right? This applies to every function in the url_aggregator

Reply:
Yes but it does not matter. Unsigned overflow is well defined in C++

Reply:
Wouldn't new_difference be too big of a number, effecting the updates of each url_components? How does that not create a bug. I think I'm missing something.

Reply:
What is `12 + 4294967295` as `uint32_t`? It is 11.

Reply:
I will add a comment. If you think there is a bug, consider adding a test case. I don't think there is a bug.

Reply:
Note also that it is not new to this PR.